sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeActionsAsyncClient.java

```diff
@@ -278,183 +441,298 @@ private JobManifestTasks getJobManifestTasks(TextAnalyticsActions actions) {
+        task.setParameters(getPiiTaskParameters(action)).setTaskName(action.getActionName());
+        return task;
+    }
+
+    private List<PiiTask> toPiiTasks(TextAnalyticsActions actions) {
         final List<PiiTask> piiTasks = new ArrayList<>();
         for (RecognizePiiEntitiesAction action : actions.getRecognizePiiEntitiesActions()) {
-            if (action == null) {
-                piiTasks.add(null);
-            } else {
-                piiTasks.add(
-                    new PiiTask()
-                        .setTaskName(action.getActionName())
-                        .setParameters(
-                            new PiiTaskParameters()
-                                .setModelVersion(action.getModelVersion())
-                                .setLoggingOptOut(action.isServiceLogsDisabled())
-                                .setDomain(PiiTaskParametersDomain.fromString(
-                                    action.getDomainFilter() == null ? null
-                                        : action.getDomainFilter().toString()))
-                                .setStringIndexType(StringIndexType.UTF16CODE_UNIT)
-                                .setPiiCategories(toCategoriesFilter(action.getCategoriesFilter()))));
-            }
+            piiTasks.add(
+                action == null
+                    ? null
+                    : new PiiTask()
+                          .setTaskName(action.getActionName())
+                          .setParameters(getPiiTaskParameters(action)));
         }
         return piiTasks;
     }
 
-    private List<KeyPhrasesTask> toKeyPhrasesTask(TextAnalyticsActions actions) {
+    private PiiTaskParameters getPiiTaskParameters(RecognizePiiEntitiesAction action) {
+        return (PiiTaskParameters) new PiiTaskParameters()
+                                       .setStringIndexType(StringIndexType.UTF16CODE_UNIT)
+                                       .setDomain(PiiDomain.fromString(
+                                           action.getDomainFilter() == null
+                                               ? null : action.getDomainFilter().toString()))
+                                       .setPiiCategories(toCategoriesFilter(action.getCategoriesFilter()))
+                                       .setModelVersion(action.getModelVersion())
+                                       .setLoggingOptOut(action.isServiceLogsDisabled());
+    }
+
+    private HealthcareLROTask toHealthcareLROTask(AnalyzeHealthcareEntitiesAction action) {
+        if (action == null) {
+            return null;
+        }
+        final HealthcareLROTask task = new HealthcareLROTask();
+        task.setParameters(getHealthcareTaskParameters(action)).setTaskName(action.getActionName());
+        return task;
+    }
+
+    private HealthcareTaskParameters getHealthcareTaskParameters(AnalyzeHealthcareEntitiesAction action) {
+        return (HealthcareTaskParameters) new HealthcareTaskParameters()
```

Comment:
this should include fhirVersion if you're exposing fhirBundle on the result

Reply:
Will create a separate PR to address it.